TripPlanner/View Controllers/TripViewController.m

```diff
@@ -0,0 +1,259 @@
+@property (weak, nonatomic) IBOutlet UIDatePicker *startDatePicker;
+@property (weak, nonatomic) IBOutlet UIDatePicker *endDatePicker;
+@property (weak, nonatomic) IBOutlet UITextField *guestsField;
+@property (weak, nonatomic) IBOutlet UILabel *guestList;
+@property (weak, nonatomic) IBOutlet UITextView *descriptionTextView;
+@property (weak, nonatomic) IBOutlet UIBarButtonItem *cancelButton;
+@property (weak, nonatomic) IBOutlet UIBarButtonItem *saveButton;
+@property (strong, nonatomic) NSMutableArray *guests;
+@property (strong, nonatomic) NSMutableArray *guestUsernames;
+
+
+@end
+
+@implementation TripViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view.
+    self.guests = [[NSMutableArray alloc] init];
+    self.guestUsernames = [[NSMutableArray alloc] init];
+    self.descriptionTextView.layer.borderWidth = 2;
+}
+
+// get guests array
+- (NSMutableArray *)getGuestsArray:(NSString *)guestsString {
+    // separate guests in guestsString
+    NSArray *guestItems = [guestsString componentsSeparatedByString:@","];
+    // strip spaces and add to a new mutable array
+    NSMutableArray *guests = [[NSMutableArray alloc] init];
+    for (NSString* guestItem in guestItems) {
+        NSString *trimmed = [guestItem stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
+        [guests addObject:trimmed];
+    }
+    
+    return guests;
+}
+
+// resize image
+- (UIImage *)resizeImage:(UIImage *)image withSize:(CGSize)size {
+    UIImageView *resizeImageView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, size.width, size.height)];
+    
+    resizeImageView.contentMode = UIViewContentModeScaleAspectFill;
+    resizeImageView.image = image;
+    
+    UIGraphicsBeginImageContext(size);
+    [resizeImageView.layer renderInContext:UIGraphicsGetCurrentContext()];
+    UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
+    UIGraphicsEndImageContext();
+    
+    return newImage;
+}
+
+// after user picks image, set image view to resized image
+- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<NSString *,id> *)info {
+    // Get the image captured by the UIImagePickerController
+    UIImage *originalImage = info[UIImagePickerControllerOriginalImage];
+    
+    // resize image
+    CGSize size = CGSizeMake(180, 250);
+    UIImage *resizedImage = [self resizeImage:originalImage withSize:size];
+    
+    // set image
+    [self.tripImageView setImage:resizedImage];
+    
+    // Dismiss UIImagePickerController to go back to original view controller
+    [self dismissViewControllerAnimated:YES completion:nil];
+}
+
+// show an image picker that allows user to choose source type
+- (void)createImagePicker:(NSString *)sourceType{
+    
+    UIImagePickerController *imagePickerVC = [UIImagePickerController new];
+    
+    if([sourceType isEqualToString:@"Camera"]) {
```

Comment:
is this the typical way of deciding on the source type? Is there a better way than string comparison, e.g. perhaps a property?